docs/caching2.md

```diff
@@ -0,0 +1,100 @@
+- Caching at the level of leveraging a [CDN](https://github.com/artsy/metaphysics/blob/1b3940a2387ab0bf308c2ca7b597e40fad015b4d/docs/cdn.md) version of Metaphysics - which has the effect of caching the entire response for a given query.
+
+For both of these options, there are several different configurations that can be used/combined to achieve desired behavior.
+
+### General Cache Behavior
+
+#### Logged in:
+
+#### Client-side caching:
+
+By default, we cache on the client for all routes _for a given session_. This means we only cache on the client, after the app mounts. (In other words, the cache isn't persisted across sessions or users.) This enables speedy transitions to pages already visited.
+
+Cache can be disabled at the route level via the `cacheConfig` prop on the route (see below), or applying `fetchPolicy: "network-only"`. Note that there are a few different types of fetch policies, which can be [read about here](https://relay.dev/docs/guided-tour/reusing-cached-data/fetch-policies/). A notable one is `store-and-network`, which will return a cache of the page and revalidate in the background, updating the page with the latest data if there's something new.
+
+#### Leveraging Metaphysics CDN:
+
+In general, any request made to the CDN with either an access token header _or_ a `Cache-Control: no-cache` header will bypass the CDN cache and hit the Metaphysics server directly. Any requests that have _neither_ of these headers will be cached by the CDN. When caching is occurring, a `max-age` value is respected by the CDN if provided. Thus, Force controls the cacheability of any given query by setting/not setting these headers appropriately, as well as including an optional `max-age` value when desired.
+
+Since logged-in queries typically always propagate their access token, they will _never_ be cached by the CDN, unless they are opted-in.
+
+For logged-in queries, there is an opt-in directive `@cacheable` that can be added to a query. When present - no access token will be propagated with the request, and the `no-cache` value will not be included - which will allow the CDN to cache this query. This is useful for queries that are not user-specific, and can be cached for all users.
+
+There is also an escape hatch - a query param in the URL: `?nocache=true` - that will force the CDN to bypass the cache and hit Metaphysics directly _despite_ a `@cacheable` directive. This can be useful for a privileged user to see an unpublished page (which we wouldn't want to cache).
+
+Setting/unsetting the `Cache-Control` header, and propagating any `max-age` is accomplished in a Relay middleware: [`System/Relay/middleware/cacheHeaderMiddleware.tsx`](System/Relay/middleware/cacheHeaderMiddleware.tsx).
+
+Un-setting the access token header for `@cacheable` queries is accomplished in a Relay middleware: [`System/Relay/createRelaySSREnvironment.ts`](System/Relay/createRelaySSREnvironment.ts).
+
+#### Logged out:
```

Comment:
#suggestion -- We should start with logged out and then move to logged in. This (gently) exposes devs to the subject. From there we can layer on more info with logged in